activerecord/lib/active_record/connection_adapters/abstract/schema_creation.rb

```diff
@@ -84,6 +84,8 @@ def visit_ForeignKeyDefinition(o)
             FOREIGN KEY (#{quote_column_name(o.column)})
               REFERENCES #{quote_table_name(o.to_table)} (#{quote_column_name(o.primary_key)})
           SQL
+
+          sql << " DEFERRABLE INITIALLY #{o.options[:deferrable].to_s.upcase}" if o.deferrable
```

Comment:
NOTE: This PR needs to be merged after https://github.com/rails/rails/pull/47659 because the return value of `ForeignKeyDefinition#deferrable` will be changed.